components/pagination/Pagination.js

```diff
@@ -0,0 +1,130 @@
+import theme from './theme.css';
+
+class Pagination extends PureComponent {
+  static propTypes = {
+    inverse: PropTypes.bool,
+    nextPageText: PropTypes.string,
+    numItems: PropTypes.number.isRequired,
+    numItemsPerPage: PropTypes.number,
+    maxNumPagesVisible: PropTypes.number,
+    currentPage: PropTypes.number.isRequired,
+    onPageChange: PropTypes.func.isRequired,
+    prevPageText: PropTypes.string,
+    className: PropTypes.string,
+  };
+
+  static defaultProps = {
+    currentPage: 1,
+    inverse: false,
+    numItemsPerPage: 10,
+    maxNumPagesVisible: 7,
+  };
+
+  constructor() {
+    super(...arguments);
+
+    this.handleChange = ::this.handleChange;
+  }
+
+  getNumPages() {
+    return Math.ceil(this.props.numItems / this.props.numItemsPerPage);
+  }
+
+  handleChange(page) {
+    this.props.onPageChange(page);
+  }
+
+  render() {
+    const { className, currentPage, inverse, maxNumPagesVisible, nextPageText, prevPageText, ...others } = this.props;
+    const numPages = this.getNumPages();
+
+    if (numPages < 2) {
+      return null;
+    }
+
+    const classNames = cx(theme['pagination'], { [theme['inverse']]: inverse }, className);
+    const iterator = [currentPage];
+    const rest = omit(others, ['numItems', 'numItemsPerPage']);
+
+    let counter = 0;
+
+    while (iterator.length < maxNumPagesVisible && iterator.length < numPages) {
+      if (counter % 2 === 0 && iterator[0] !== 1) {
+        iterator.unshift(iterator[0] - 1);
+      } else if (counter % 2 === 1 && iterator[iterator.length - 1] !== numPages) {
+        iterator.push(iterator[iterator.length - 1] + 1);
+      }
+
+      counter++;
+    }
+
+    // make sure the first and the last item are the first and the last page
+    iterator[0] = 1;
+    iterator[iterator.length - 1] = numPages;
+
+    // should we replace the 2nd and 2nd last by ellipsis?
+    if (iterator.length > 1 && iterator[1] !== 2) {
+      iterator[1] = 'ellipsis-left';
+    }
+    if (iterator.length > 1 && iterator[iterator.length - 2] !== numPages - 1) {
+      iterator[iterator.length - 2] = 'ellipsis-right';
+    }
+
+    return (
+      <Box data-teamleader-ui="pagination" className={classNames} element="nav" {...rest}>
+        <ul className={theme['list']}>
+          {currentPage > 1 && (
+            <li className={theme['list-item']}>
+              <LinkButton
+                icon={<IconChevronLeftMediumOutline />}
+                inverse={inverse}
+                label={prevPageText}
+                onClick={() => this.handleChange(currentPage - 1)}
+              />
+            </li>
+          )}
+          {iterator.map(page => {
+            return (
+              <li key={page} className={theme['list-item']}>
+                {('' + page).indexOf('ellipsis') === -1 ? (
+                  <LinkButton
+                    className={page === currentPage ? theme['current'] : ''}
+                    inverse={inverse}
+                    label={page}
```

Comment:
``label={$page}`` will remove the warning for the label prop not being a string. Or you could also allow numbers for the label in `LinkButton` of course

Reply:
I updated LinkButton and so allowed label to be a number as well.